src/components/Documents.tsx

```diff
@@ -16,43 +43,42 @@ function Documents() {
       background="#FFFFFF"
     >
       <GridItem>
-        <Image src="/blue.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
-      </GridItem>
-      <GridItem>
-        <Image src="/grey_rectangle.svg" alt="noimage" />
+        <Image src="/blue.svg" alt="noimage" onClick={onOpen} />
+        <Modal isOpen={isOpen} onClose={onClose}>
+          <ModalOverlay />
+          <ModalContent>
+            <ModalHeader>New Document</ModalHeader>
+            <ModalCloseButton />
+            <ModalBody>
+              {" "}
+              <Input
+                value={text}
+                onChange={handleChange}
+                placeholder="Enter Text"
+              />{" "}
+            </ModalBody>
+
+            <ModalFooter>
+              <Button
+                colorScheme="blue"
+                mr={3}
+                onClick={() => {
+                  increase();
+                  onClose();
```

Comment:
In this case, you wanna make sure that onClose() is executed after increase(). But increase() involves updating state, which is asynchronous. So, what you could do is call increase(), but also onClose() run in useEffect() by checking for state updates to the state array of documents. So something like this:

`onClick={() => {
                  increase();
  `

`useEffect(()=> {
     // Have a check here to close the modal whenever that button is clicked - could have some kind of stateful value bound to that?
    onClose();
}, []);`

Reply:
In this case, you don't have to - but if `onClose()` relies on a stateful change from the previous `increase()`, then you would need to! Here is a good [article](https://medium.com/@sanjeewafern/perform-an-immediate-action-after-setting-state-on-a-state-variable-with-react-functional-component-e9e1526f655a) that explains this.